### PR TITLE
[APR-229] Add named outputs for Saluki events

### PR DESCRIPTION
**Pull request by rayz**
# Context
Currently, a single `EventBuffer` is used to hold metrics, events, and service checks. We would like each event type to have their own named outputs.

# Solution
We use 2 additional `EventBuffer` for events and service checks, leaving the original event buffer dedicated to metrics. 

We also add a new blackhole destination for events and service checks, which we will remove when we have a destination that supports them (next pr  #126)

# Testing
Should not change anything for metrics.
Run `make run-dsd-basic-udp` and see metrics.

[Example for sending service checks
](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=servicechecks)

[Example for sending events
](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=events)

Blackhole will log that it has received events when sending events/service checks.

### CODE REVIEW

File: lib/saluki-components/src/sources/dogstatsd/mod.rs [hunk 1]
```diff
@@ -257,9 +259,16 @@ impl SourceBuilder for DogStatsDConfiguration {
     }
 
     fn outputs(&self) -> &[OutputDefinition] {
-        static OUTPUTS: &[OutputDefinition] = &[OutputDefinition::default_output(DataType::Metric)];
-
-        OUTPUTS
+        static OUTPUTS: Lazy<Vec<OutputDefinition>> = Lazy::new(|| {
```
Comment (rayz) [inline lib/saluki-components/src/sources/dogstatsd/mod.rs:262]: `named_output` is not const so I used `once_cell` per the editor's suggestion in order to update `OUTPUTS`

File: lib/saluki-core/src/topology/interconnect/event_buffer.rs [hunk 2]
```diff
@@ -115,6 +141,22 @@ impl Iterator for IntoIter {
...
+/// Predicate function for extracting eventd events.
+pub fn is_eventd(event: &Event) -> bool {
+    matches!(event.data_type(), DataType::EventD)
+}
+
+/// Predicate function for extracting service check events.
+pub fn is_service_check(event: &Event) -> bool {
+    matches!(event.data_type(), DataType::ServiceCheck)
+}
+
```
Comment (tobz) [inline lib/saluki-core/src/topology/interconnect/event_buffer.rs:159]: I would just move these directly to `Event` and have them match on the structure itself (i.e. `matches!(self, Event::EventD(_))`).

Simple enough as a basic helper method and this module is kind of an obscure place for them to leave, even if we primarily use them here.

**Review by tobz — CHANGES_REQUESTED** [review]

File: lib/saluki-components/src/sources/dogstatsd/mod.rs (hunk 1 above)
Comment (tobz) [reply lib/saluki-components/src/sources/dogstatsd/mod.rs:263]: Actually, thinking about it... I'd also be fine if we bumped to Rust 1.80.0 (just change the value in `rust-toolchain.toml`) and used `std::sync::LazyLock`, to keep the slightly cleaner closure-in-variable-declaration pattern that we're using `Lazy` to achieve.

File: lib/saluki-event/src/lib.rs [hunk 3]
```diff
@@ -120,4 +120,20 @@ impl Event {
             _ => None,
         }
     }
+
+    #[allow(unused)]
+    /// Predicate function for extracting metric events.
```
Comment (tobz) [inline lib/saluki-event/src/lib.rs:125]: I would reword these as ``Returns `true` if the event is a metric``.

**Review by tobz — CHANGES_REQUESTED** [review]
Just one small change and then I think this is good to go.

**Review by tobz — APPROVED** [review]
